syncer/checkpoint.go

```diff
@@ -706,13 +756,48 @@ func (cp *RemoteCheckPoint) genUpdateSQL(cpSchema, cpTable string, binlogName st
 	if len(tiBytes) == 0 {
 		tiBytes = []byte("null")
 	}
-	args := []interface{}{cp.id, cpSchema, cpTable, binlogName, binlogPos, tiBytes, isGlobal}
+	args := []interface{}{cp.id, cpSchema, cpTable, location.Position.Name, location.Position.Pos, location.GTIDSet.String(), tiBytes, isGlobal}
 	return sql2, args
 }
 
-func (cp *RemoteCheckPoint) parseMetaData() (*mysql.Position, error) {
+func (cp *RemoteCheckPoint) parseMetaData() (*pbinlog.Location, error) {
 	// `metadata` is mydumper's output meta file name
 	filename := path.Join(cp.cfg.Dir, "metadata")
 	cp.logCtx.L().Info("parsing metadata from file", zap.String("file", filename))
-	return utils.ParseMetaData(filename)
+	pos, gsetStr, err := utils.ParseMetaData(filename)
+	if err != nil {
+		return nil, err
+	}
+
+	gset, err := gtid.ParserGTID(cp.cfg.Flavor, gsetStr)
+	if err != nil {
+		return nil, err
+	}
+
+	return &pbinlog.Location{
+		Position: *pos,
+		GTIDSet:  gset,
+	}, nil
+}
+
+func minLocation(flavor string) pbinlog.Location {
+	if flavor == mysql.MySQLFlavor {
+		return pbinlog.Location{
+			Position: minPosition,
+			GTIDSet:  minGTIDSet(flavor),
+		}
+	}
+
```

Comment:
It seems these lines can be removed.

Reply:
removed in e2e705eda2bc1db71795a9eeb7226731e5c64e8d